switchio/connection.py

```diff
@@ -52,12 +54,79 @@ def run_in_order_threadsafe(awaitables, loop, timeout=0.5, block=True):
     return future
 
 
-class AsyncIOConnection(object):
+async def try_connect(host, port, password, prot, loop, log):
+    """Try to create a connection and authenticate to the
+    target FS ESL.
+    """
+    msg = ("Failed to connect to server at '{}:{}'\n"
+           "Please check that FreeSWITCH is running and "
+           "accepting ESL connections.".format(host, port))
+    for _ in range(5):
+        try:
+
+            await asyncio.wait_for(
+                loop.create_connection(lambda: prot, host, port),
+                timeout=0.1)
```

Comment:
Is that maybe a bit fast?

Reply:
Turns out, no, otherwise it wouldn't work but that's in local testing of course ;)
Good point that this might be a problem when making remote attempts.